lib/offsite_payments/integrations/pay_vector.rb

```diff
@@ -0,0 +1,776 @@
+#!/bin/env ruby
+# encoding: utf-8
+
+module OffsitePayments #:nodoc:
+  module Integrations #:nodoc:
+    module PayVector
+
+      mattr_accessor :service_url
+      self.service_url = 'https://mms.iridiumcorp.net/Pages/PublicPages/PaymentForm.aspx'
+
+      def self.notification(post, options = {})
+        Notification.new(post, options)
+      end
+
+      class Helper < OffsitePayments::Helper
+        
+        # Replace with the real mapping
+        mapping :account, "MerchantID"
+        mapping :order, 'OrderID'
+        mapping :currency, 'CurrencyCode'
+        mapping :transaction_type, 'TransactionType'
+        
+        mapping :customer, :email => 'EmailAddress',
+                           :phone => 'PhoneNumber'
+
+        mapping :billing_address, :city     => 'City',
+                                  :address1 => 'Address1',
+                                  :address2 => 'Address2',
+                                  :state    => 'State',
+                                  :zip      => 'PostCode',
+                                  :country  => 'CountryCode'
+
+        mapping :notify_url, 'ServerResultURL'
+        mapping :return_url, 'CallbackURL'
+        mapping :description, 'OrderDescription'
+        
+         # Fetches the md5secret and adds MERCHANT_ID and API TYPE to the form
+        def initialize(order, account, options = {})
+          #store merchant password and the preSharedKey as private variables so they aren't added to the form
+          @merchant_password = options.delete(:credential2)
+          @pre_shared_key = options.delete(:credential3)
+          #currency is given as 3 char code - so convert it to ISO code
+          options[:currency] = convert_currency_short_to_ISO_code(options[:currency])
+          order = order.to_s
+          options.each do |key, option|
+            options[key] = option.to_s
+          end
+          super
+
+          add_field("OrderDescription", "ActiveMerchant Order " + order)
+          if(options.has_key?(:amount))
+            add_field("Amount", minor_currency_from_major(options[:amount], options[:currency]))
+          end
+
+          transaction_date_time
+          populate_fields_with_defaults
+        end
+        
+        def minor_currency_from_major(amount, currency_iso_code)
```

Comment:
We use the [`money`](https://github.com/RubyMoney/money) gem, you should leverage where possible